community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/executor/TaskExecutor.java

```diff
@@ -46,23 +43,31 @@
     /**
      * Shuts down this {@link TaskExecutor}, disallowing new tasks to be {@link #submit(Task) submitted}.
      *
-     * @param flags {@link #SF_AWAIT_ALL_COMPLETED} will wait for all queued or already executing tasks to be
-     * executed and completed, before returning from this method. {@link #SF_ABORT_QUEUED} will have
-     * submitted tasks which haven't started executing yet cancelled, never to be executed.
+     * submitted tasks will be processed before returning from this method.
+     */
+    void shutdown();
+
+    /**
+     * Puts this executor into panic mode. Call to {@link #shutdown()} has no effect after a panic.
+     * Call to {@link #assertHealthy()} will communicate the panic as well.
+     * This semantically includes something like a shutdown, but submitted tasks which haven't started
+     * being processed will be aborted and currently processing tasks will not be awaited for completion.
+     *
+     * @param panic cause of panic.
      */
-    void shutdown( int flags );
+    void panic( Throwable panic );
 
     /**
-     * @return {@code true} if {@link #shutdown(int)} has been called, otherwise {@code false}.
+     * @return {@code true} if {@link #shutdown()} has been called, otherwise {@code false}.
```

Comment:
From  this comment it is not clear to be if `isShutdown` will return true or false if `panic` has been called. Can we please clarify this?